Issue#50 deploy infrastructure on docker

Romanov-Niko opened the pull request:
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

Commit:
Issue#94-Fixed README.md

Commit:
Issue$94-Fixed Roles' package

Commit:
Issue#94-Fixed project's documentation

Commit:
Issue#96-Fixed UserApiServiceImpl and NewsApiServiceImpl

Commit:
Issue#96-Edited some comments in classes and added todo tasks

Commit:
Issue#94-Fixed some mistakes in documentation

Commit:
Merge pull request #95 from ita-social-projects/Issue#94-Fixed-Readme-And-Project-Documentation

Issue#95 fixed readme and project documentation

Commit:
Merge branch 'dev' into Issue#96-Configure-Liquibase-Migration

Commit:
Issue#96-Fixed comments in NewsApiServiceImpl and UserApiServiceImpl

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Created database and application creation in containers with docker-maven-plugin.

Commit:
Issue#96-Moved admin's credentials to properties' file

Commit:
Issue$96-Cleaned exceptions package

Commit:
Issue$96-Fixed some comments in code

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Moved database and application creation in a separate profile.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Added link to the database in container, renamed database container name in docker-compose and added phase of build and start in docker-maven-plugin.

Commit:
Issue#96-Deleted dependency's duplicates in pom.xml of home-application

Commit:
Issue$96-Fixed DataLoader with admin's credentials in property file

Commit:
Issue$96-Moved configuration of liquibase in propertyfile, cleaned pom file and prepared for liquibase based fillin db

Commit:
Issue$96-Fixed documentaion of Liquibase

Commit:
Issue$96-Rewrote liquibase-migration, moved admin's credentials in liquibase.properties

Commit:
Issue#96-Fixed Liquibase's documentation

Commit:
Issue#96-Fixed some grammatical mistakes

Commit:
Issue#96-Fixed some grammatical mistakes#2

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactor docker profile

Moved docker profile from home-application to home-api-tests.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactored docker profile

Added properties(postgres, jar names and version, containers ports) and links to them.

Abhai2016 commented on home-data/src/main/resources/application.properties:
```diff
@@ -1,6 +1,8 @@
 spring.datasource.driver-class-name=org.postgresql.Driver
 spring.jersey.type=filter
 server.port=${PORT:8080}
+spring.jpa.hibernate.ddl-auto=create-drop
```
#spring.datasource.url=${DATASOURCE_URL:jdbc:postgresql://db-postgres:5432/postgres}
spring.jpa.hibernate.ddl-auto=create-drop 

Эти строки по сути являются кастылем, поэтому они не должны быть на удаленном репозитории.

Romanov-Niko replied:
Вот этот датасорс который закомментирован, я использую для того, чтобы приложение развернулось в докере. Без него оно по локалхосту не найдет контейнер с бд.

Romanov-Niko replied:
Когда кто-то захочет запустить тесты, надо будет поменять датасорс

Romanov-Niko replied:
Насчет create-drop согласен, он там не надо, всё исправил.

Commit:
Issue#96-Finally git rid of DataLoader

Commit:
Issue#96-Fixed some constraints and name of id in changeSets

Commit:
Issue#96-Fixed some comments in code

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Removed hibernate ddl from properties

PoLytvy committed:
Issue#50-added-test-class

Commit:
Issue#96-Ecnrypted and moved user's credentials in changeLog

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
 
+    <properties>
+        <fabric8io.docker.maven.plugin.version>0.34.1</fabric8io.docker.maven.plugin.version>
+        <docker.jdk.version>openjdk:11-jdk-buster</docker.jdk.version>
```
Please add to the property name a context of the image.

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+        <postgres.db>postgres</postgres.db>
+        <postgres.user>postgres</postgres.user>
+        <postgres.password>password</postgres.password>
+        <postgres.url>jdbc:postgresql://db-postgres:5432/postgres</postgres.url>
```
`jdbc:postgresql://${db.container.name}:${db.internal.port}/${postgres.db}`

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+        <postgres.password>password</postgres.password>
+        <postgres.url>jdbc:postgresql://db-postgres:5432/postgres</postgres.url>
+        <home.application.jar.name>home-application</home.application.jar.name>
+        <home.application.jar.version>0.0.1-SNAPSHOT</home.application.jar.version>
```
<home.application.jar.version>${revision}</home.application.jar.version>

Romanov-Niko replied:
Done

Commit:
Issue#96-Got back LiquibaseRunne with old realization

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+        <home.data.migration.jar.version>0.0.1-SNAPSHOT</home.data.migration.jar.version>
+        <postgres.port>5433:5432</postgres.port>
+        <liquibase.port>8000:8000</liquibase.port>
+        <home.application.port>8080:8080</home.application.port>
```
localhost:${app.external.port}

Romanov-Niko replied:
Done

Commit:
Issue$96-Got back old documentation of liquiebase in README

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+                                            <link>db-postgres</link>
+                                        </links>
+                                        <wait>
+                                            <log>Successfully released change log lock</log>
```
Let's fail the build if the scripts are running too long.

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+                                            <port>${home.application.port}</port>
+                                        </ports>
+                                        <links>
+                                            <link>db-postgres</link>
```
Please add wait on the healthy app.

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+                                    <goal>build</goal>
+                                    <goal>start</goal>
+                                </goals>
+                            </execution>
```
Please add execution for stopping the app when tests are completed.

Romanov-Niko replied:
Done

kh0ma commented on home-data/src/main/resources/application.properties:
```diff
@@ -1,6 +1,7 @@
+#spring.datasource.url=${DATASOURCE_URL:jdbc:postgresql://db-postgres:5432/postgres}
```
set DATASOURCE_URL=
java -jar -Dspring.datasource.url=

java -jar -Dapplication.properties.path=/asdasd/asdasd/

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+
+    <build>
+        <plugins>
+            <plugin>
```
Please move to profile

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,195 @@
+                                    <version>${home.data.migration.jar.version}</version>
+                                    <type>jar</type>
+                                    <overWrite>true</overWrite>
+                                    <outputDirectory>${project.build.directory}</outputDirectory>
```
Make a property with path for docker artifacts

Romanov-Niko replied:
Done

kh0ma commented on home-data/src/main/resources/application.properties:
```diff
@@ -1,6 +1,7 @@
 spring.datasource.driver-class-name=org.postgresql.Driver
 spring.jersey.type=filter
-server.port=${PORT:8080}
+server.port=${PORT:8080}
```
Don't forget to provide port

Romanov-Niko replied:
Done

Commit:
Issue$96-Moved properties from property file to pom.xml

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactored docker profile

Added new properties and links to them, moved maven dependency plugin into the docker profile, changed artifacts directory from target/ to target/docker/artifacts.

PoLytvy committed:
Issue#50: Fixed client pom.mustache file

kh0ma commented on docker-compose.yml:
```diff
@@ -1,11 +1,10 @@
       POSTGRES_PASSWORD: password
     ports:
-      - 5433:5432
+      - 5433:5432
```
Please do not forget to add newline at the end of the file. It can affect git diff in future changes.

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -13,4 +13,223 @@
+                            </images>
+                        </configuration>
+                        <executions>
+                            <execution>
```
What is about to check if the containers with the same name as we are going to run are running and stop them before?

Romanov-Niko replied:
Done

Commit:
Issue#96-Added a new line in the end of class

Commit:
Issue#96-Deleted user from changeLogFile so only admin left in data-migration

Romanov-Niko committed:
Added new line at the end in docker compose

Commit:
Issue#96-Fixed mistype in Issue96.xml

Commit:
Issue#96-Added jpa validation for matching schema of db and entities

Commit:
Issue#96-Forgot about new line in the end of the file

Commit:
ISsue#96-Got back driver of db in the pox.xml

Commit:
Issue#96-Fixed ids of changeSets

PoLytvy committed:
Issue#104: Updated scripts according to new DB schema and added sequence

PoLytvy committed:
Issue#50: added NewsApiTest

Commit:
Merge pull request #99 from ita-social-projects/Issue#96-Configure-Liquibase-Migration

Issue#96 configure liquibase migration

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Random port and stop existing containers

Added plugin that allows us to use available ports and prevent collisions. Added stop goal to execute section of docker-maven-plugin to stop existing containers.

Romanov-Niko committed:
Merge remote-tracking branch 'origin/Issue#50-Deploy-infrastructure-on-docker' into Issue#50-Deploy-infrastructure-on-docker

# Conflicts:
#	docker-compose.yml

kh0ma committed:
Add heroku deploy badge

PoLytvy committed:
Issue#50: configured plagins in pom file

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added profile to github actions

Added docker profile to github actions and changed version from 0.34.1 to 0.24.0 to fix bug with stopping of build.

PoLytvy committed:
Issue#50: Updated API tests, corrected openapi file

Commit:
Merge pull request #111 from ita-social-projects/add_heroku_deploy_badge

Add heroku deploy badge

Commit:
Fixed badge of heroku's build

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added data migration module building in github actions

Added separate building of data migration module on github actions, cause there was problems with retrieving jar from unbuilt module with maven-dependency-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added dependency on data migration module

Fixed problem with maven dependency plugin, now we don't need to build data-migration module separately

PoLytvy committed:
Merge branch 'dev' into Issue#104-update-database-tables

# Conflicts:
#	home-data-migration/src/main/resources/db/changelog/0.0.1/Issue33.xml

PoLytvy committed:
Issue#104: Updated scripts according to DB schema and added new sequence

PoLytvy committed:
Issue#104: Fixed id type

Romanov-Niko committed:
Merge pull request #105 from ita-social-projects/Issue#104-update-database-tables

Issue#104: Updated scripts according to new DB schema and added sequence

Commit:
Merge branch 'dev' into fixHerokuBadge

alexorlenko committed:
Merge pull request #114 from ita-social-projects/fixHerokuBadge

Fixed badge of heroku's build

PoLytvy committed:
Issue#104: Made test profile and finished delete test

Romanov-Niko committed:
Merge pull request #113 from ita-social-projects/Issue#50-Introduce-Api-tests

Issue#50 introduce api tests

Romanov-Niko committed:
Merge remote-tracking branch 'origin/Issue#50-Deploy-infrastructure-on-docker' into Issue#50-Introduce-Api-test-framework

Romanov-Niko committed:
Issue#50-Deploy-Infrastructure-on-docker/Added tests

Romanov-Niko committed:
Merge branch 'dev' of https://github.com/ita-social-projects/Home into Issue#50-Deploy-infrastructure-on-docker

Romanov-Niko committed:
Issue#50-Deploy-Infrastructure-on-docker/Added profile with tests to github actions
